manticore/core/workspace.py

```diff
@@ -454,13 +454,6 @@ def save_testcase(self, state, prefix, message=''):
 
         self._named_key_prefix = prefix
         self._increment_id()
-
-        self.save_summary(state, message)
```

Comment:
this should not be removed?

Reply:
What to save have nothing to do with the workspace. Not sure where I put it.

Reply:
workspace directories look pretty empty, not sure it was put somewhere else

```
[I] mark ubuntu /m/h/c/m/e/linux (dev-evm) ❯ manticore basic
2017-09-13 18:46:38,504: [99283] MANTICORE:INFO: Loading program basic
2017-09-13 18:46:39,373: [99283] MANTICORE:INFO: Starting 1 processes.
2017-09-13 18:46:43,445: [99283] MANTICORE:INFO: Generated testcase No. 0 - Program finished with exit status: 0L
2017-09-13 18:46:44,429: [99283] MANTICORE:INFO: Generated testcase No. 1 - Program finished with exit status: 0L
2017-09-13 18:46:44,443: [99283] MANTICORE:INFO: Results in /mnt/hgfs/code/manticore/examples/linux/mcore_7zPYZL
2017-09-13 18:46:44,444: [99283] MANTICORE:INFO: Instructions executed: 12464
2017-09-13 18:46:44,444: [99283] MANTICORE:INFO: Coverage: 3143 different instructions executed
2017-09-13 18:46:44,444: [99283] MANTICORE:INFO: Total time: 5.07086396217
2017-09-13 18:46:44,444: [99283] MANTICORE:INFO: IPS: 2457
[I] mark ubuntu /m/h/c/m/e/linux (dev-evm) ❯ l /mnt/hgfs/code/manticore/examples/linux/mcore_7zPYZL
total 864
-rw-r--r-- 1 501 dialout     37 Sep 13 18:46 command.sh
-rw-r--r-- 1 501 dialout 424234 Sep 13 18:46 test_00000000.pkl
-rw-r--r-- 1 501 dialout 424469 Sep 13 18:46 test_00000001.pkl
-rw-r--r-- 1 501 dialout  34573 Sep 13 18:46 visited.txt
```

Reply:
Hmm ideas on where to put them? We do not want (for example)  test_000001.stdin for EVM

Reply:
In general, the platform somewhere... platform has a method returning a dict of X -> file content, where X is what is now passed to self._named_stream()?

Reply:
paging @yan on this also

Reply:
I'm not sure Platform needs to return a dict that a Workspace can serialize, it should probably implement a `dump_summary(workspace, state)`  method and serialize what needs to be serialized.

Reply:
I think if that function returns a dict, it will make for easier testing in per platform tests.

Since this change will take place in `ManticoreOutput.save_testcase`, the invocation would be `state.platform.dump_summary(self)`, which seems slightly code smelly